pkgs/tools/networking/xray/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib, fetchFromGitHub, fetchurl, symlinkJoin, buildGoModule, runCommand, makeWrapper, nixosTests
+, v2ray-geoip, v2ray-domain-list-community, assets ? [ v2ray-geoip v2ray-domain-list-community ]
+}:
+
+let
+  version = "1.5.5";
+
+  src = fetchFromGitHub {
+    owner = "XTLS";
+    repo = "Xray-core";
+    rev = "v${version}";
+    sha256 = "x2aNAu+H3qJIKjQbE0rYxfQAWucvOMaU2eSy9YIZdcQ=";
+  };
+
+  vendorSha256 = "sha256-PBcUKUP06QKTu3djxoTb7B7Ld86fZz7+SDewRZy1YmQ=";
+
+  assetsDrv = symlinkJoin {
+    name = "v2ray-assets";
+    paths = assets;
+  };
+
+in
+buildGoModule rec {
+  pname = "v2ray";
+  inherit version src;
+  inherit vendorSha256;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  doCheck = false;
+
+  buildPhase = ''
+    buildFlagsArray=(-v -p $NIX_BUILD_CORES -ldflags="-s -w -buildid=")
+    runHook preBuild
+    go build "''${buildFlagsArray[@]}" -o xray ./main
+    runHook postBuild
+  '';
```

Comment:
Can we use the standard buildPhase?

Reply:
I think it might be fine, but I haven't tested it yet

Reply:
Please test it

Reply:
Just did the necessary changes for this